sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -121,46 +121,16 @@ namespace Azure { namespace Security { namespace Attestation {
      */
     virtual ~AttestationClient() = default;
 
-    /** @brief Construct a new Attestation Client object
-     *
-     * @param endpoint The URL address where the client will send the requests to.
-     * @param credential The authentication method to use (required for TPM attestation).
-     * @param options The options to customize the client behavior.
-     */
-    explicit AttestationClient(
-        std::string const& endpoint,
-        std::shared_ptr<Core::Credentials::TokenCredential const> credential,
-        AttestationClientOptions options = AttestationClientOptions());
-
-    /** @brief Construct a new anonymous Attestation Client object
-     *
-     * @param endpoint The URL address where the client will send the requests to.
-     * @param options The options to customize the client behavior.
-     *
-     * @note TPM attestation requires an authenticated attestation client.
-     */
-    explicit AttestationClient(
-        std::string const& endpoint,
-        AttestationClientOptions options = AttestationClientOptions())
-        : AttestationClient(endpoint, nullptr, options)
-    {
-    }
-
     /** @brief Construct a new Attestation Client object from an existing attestation client.
      *
      * @param attestationClient An existing attestation client.
      */
-    explicit AttestationClient(AttestationClient const& attestationClient)
+    AttestationClient(AttestationClient const& attestationClient)
```

Comment:
public, keep `explicit` ?

Reply:
It's the copy constructor and it can't be explicit (otherwise you can't return a `std::unique_ptr<AttestationClient>` from a method). I'm not sure why that is, but it is.

Reply:
Ok, this is weird. If I make the copy constructor explicit, I get:
```
  C:\PROGRA~1\MIB055~1\2022\ENTERP~1\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe  /nologo /TP -DAZ_RTTI -DBUILD_CURL_HTTP_TRANSPORT_ADAPTER -DCURL_STATICLIB -IG:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\inc -IG:\Az\LarryO\azure-sdk-for-cpp\sdk\core\azure-core\inc -external:I G:\Az\LarryO\azure-sdk-for-cpp\out\build\x64-DebugWithTests\vcpkg_installed\x64-windows-static\include -external:W0 /DWIN32 /D_WINDOWS /W4 /GR /EHsc /MDd /Zi /Ob0 /Od /RTC1 /permissive- /W4 /WX /wd5031 /wd4668 /wd4820 /wd4255 /wd4710 -std:c++14 /showIncludes /Fosdk\attestation\azure-security-attestation\CMakeFiles\azure-security-attestation.dir\src\attestation_client.cpp.obj /Fdsdk\attestation\azure-security-attestation\CMakeFiles\azure-security-attestation.dir\azure-security-attestation.pdb /FS -c G:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\src\attestation_client.cpp
G:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\src\attestation_client.cpp(228): error C2440: 'return': cannot convert from 'Azure::Security::Attestation::AttestationClient' to 'Azure::Security::Attestation::AttestationClient'
  G:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\src\attestation_client.cpp(228): note: Constructor for class 'Azure::Security::Attestation::AttestationClient' is declared 'explicit'
G:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\src\attestation_client.cpp(243): error C2440: 'return': cannot convert from 'const Azure::Security::Attestation::AttestationClient' to 'Azure::Security::Attestation::AttestationClient'
  G:\Az\LarryO\azure-sdk-for-cpp\sdk\attestation\azure-security-attestation\src\attestation_client.cpp(243): note: Constructor for class 'Azure::Security::Attestation::AttestationClient' is declared 'explicit'
```
However if I put a breakpoint, the copy constructor is never called.

Reply:
Digging more. [Making a copy constructor explicit prevents it from being used as a parameter to a method](https://en.wikipedia.org/wiki/Copy_constructor_(C++)#Explicit_copy_constructor). And it also means you can never return an instance of the object from a function.